File: traveller/dice.py
#!/usr/bin/python


from random import sample


def sample1(rng):
    return sample(rng,1)[0]


def die(sides):
    R = range(1,sides+1)
    def roll(dice):
        return sum(sample1(R) for i in range(dice))
    return roll

d100 = die(100)
d16 = die(16)
d6 = die(6)
d5 = die(5)
d3 = die(3)

d66 = lambda: (d6(1)*10) + d6(1)
d6_drop1 = lambda N: sum((d5(1) + 1) for n in range(N))


if __name__ == "__main__":

#   D = [d6(2), d6(2), d6(2), d6(2), d6(2), d6(2)]

#   print "UPP: %x%x%x%x%x%x [%d]" % tuple(D+[sum(D)])

    print d6(1), d6(1), d6(1), d6(1), d6(1), d6(1)

File: app/src/generators/dice.js
export function rollDie(rng, sides) {
  return Math.floor(rng() * sides) + 1;
}

export function rollDice(rng, dice, sides) {
  let total = 0;
  for (let index = 0; index < dice; index += 1) {
    total += rollDie(rng, sides);
  }
  return total;
}

export function d3(rng, dice = 1) {
  return rollDice(rng, dice, 3);
}

export function d6(rng, dice = 1) {
  return rollDice(rng, dice, 6);
}

export function d16(rng, dice = 1) {
  return rollDice(rng, dice, 16);
}

export function d66(rng) {
  return d6(rng, 1) * 10 + d6(rng, 1);
}
